CommonData/column.cpp

```diff
@@ -1811,20 +1859,54 @@ Json::Value Column::serialize() const
 
 	json["customEmptyValues"]	= _emptyValues->toJson();
 
-	json["labels"]				= jsonLabels;
+	json["labels"]				= serializeLabels();
 	json["dbls"]				= jsonDbls;
 	json["ints"]				= jsonInts;
 
 	return json;
 }
 
+Json::Value	Column::serializeLabels() const
+{
+	Json::Value jsonLabels(Json::arrayValue);
+	for (const Label* label : _labels)
+		jsonLabels.append(label->serialize());
+	
+	return jsonLabels;
+}
+
+void Column::deserializeLabels(const Json::Value & labels)
+{
+	for (Label* label : _labels)
+	{
+		_labelByIntsIdMap.erase(label->intsId());
+		label->dbDelete();
+		delete label;
+	}
+	_labelByIntsIdMap.clear();
+	_labels.clear();
+	labelsTempReset();
+
+	if (labels.isArray())
+		for (const Json::Value& labelJson : labels)
+			labelsAdd(
+						labelJson["intsId"]			.asInt(), 
+						labelJson["label"]			.asString(), 
+						labelJson["filterAllows"]	.asBool(), 
+						labelJson["description"]	.asString(), 
+						labelJson["originalValue"], 
+						labelJson["order"]			.asInt(), 
+						-1
+			);
+}
+
```

Comment:
Should these updates not be done inside a batch?

Reply:
Good one